File: php/graphql/Schema.php
<?php
/**
 * User: WindomZ
 * Date: 17-7-20
 */

namespace GraphQLExamples\Schema;

require_once __DIR__ . '/../graphql/field/Hello.php';
require_once __DIR__ . '/../graphql/field/Calc.php';
require_once __DIR__ . '/../graphql/field/User.php';

use GraphQL\Type\Definition\ObjectType;
use GraphQL\Schema;
use GraphQLExamples\Field;

function getSchema()
{
    return new Schema([
        'query' => new ObjectType([
            'name' => 'Query',
            'fields' => [
                'hello' => Field\Hello::getHelloField(),
                'bye' => Field\Hello::getByeField(),
                'user' => Field\User::getUserField(),
            ],
        ]),
        'mutation' => new ObjectType([
            'name' => 'Calc',
            'fields' => [
                'sum' => Field\Calc::getSumField(),
            ],
        ]),
    ]);
}

File: go/graphql/schema.go
package graphql_examples

import (
	"fmt"

	field "github.com/WindomZ/graphql-examples/go/graphql/field"
	"github.com/graphql-go/graphql"
)

var ExampleSchema graphql.Schema

func init() {
	var err error
	ExampleSchema, err = graphql.NewSchema(
		graphql.SchemaConfig{
			Query: graphql.NewObject(
				graphql.ObjectConfig{
					Name: "Query",
					Fields: graphql.Fields{
						"hello": field.HelloField,
						"bye":   field.ByeField,
						"user":  field.UserField,
					},
				},
			),
			Mutation: graphql.NewObject(
				graphql.ObjectConfig{
					Name: "Calc",
					Fields: graphql.Fields{
						"sum": field.SumField,
					},
				},
			),
		},
	)
	if err != nil {
		panic(fmt.Sprintf("failed to create new schema, error: %v", err))
	}
}
